ckan/lib/helpers.py

```diff
@@ -1126,8 +1127,25 @@ def _make_menu_item(menu_item, title, **kw):
 
 
 @core_helper
-def default_group_type():
-    return str(config.get('ckan.default.group_type', 'group'))
+def default_group_type(type_='group'):
+    """Get default group/organization type for using site-wide.
+    """
+    return str(config.get('ckan.default.{}_type'.format(type_), type_))
+
+
+@core_helper
+def humanize_entity_type(type_):
+    """Convert machine-readable representation of package/group type into
+    human-readable form.
+
+    Example::
+
+        >>> humanize_entity_type('group')
+        'group'
+        >>> humanize_entity_type('custom_group')
+        'custom group'
+    """
+    return type_.replace('_', ' ')
```

Comment:
```suggestion
def humanize_entity_type(entity_type, object_type, purpose):
    """Convert machine-readable representation of package/group type into
    human-readable form.

    Example::

        >>> humanize_entity_type('group', 'custom group', 'add link')
        'Add Custom Group'
        >>> humanize_entity_type('group', 'custom_group', 'breadcrumb')
        'Custom Groups'
    """
    if entity_type == object_type:
        return  # use the default text included in template
    
    name = object_type.replace('_', ' ').capitalize()
    if purpose == 'add link':
        return _('Add {object_name}').format(object_name = name)
    # ... other purposes with custom text
    return name + 's'
```

Reply:
I think it's better to *not* have an exhaustive list of translations for each object_type + purpose. If a site needs to override specific translations they can do that by overriding this helper (and they get their translation text in code that can be picked up by `extract_messages` too)

Reply:
Yep, I like this approach.